site/src/Service/NotificationProducerService.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Service;
+
+use OldSound\RabbitMqBundle\RabbitMq\ProducerInterface;
+use App\Util\NotificationHydrator;
+
+/**
+ * Class NotificationProducer
+ *
+ * This class publishes a Notification to the RabbitMQ queue
+ * which is then taken up by the Consumers
+ *
+ * @author Aquib Baig <aquibbaig97@gmail.com>
+ */
+class NotificationProducerService
```

Comment:
This class seems to be modeled after QueueDispatcherService, which isn't actually a particularly good design and needs to be fixed now that we have more than one producer service.

Please remove this class, you can inject the auto-generated service (i.e. the one you'd pass here as $notificationProducer) directly into the class method/class that needs to produce messages. (ProjectController.php in this case). Please don't use the `ProducerInterface`, but the actual class -- all producers conform to this interface, and autowire cannot really figure out which service class is the right one to bind now that there are two options.

Reply:
So, while publishing a message, do we need to get the producer just as 'this->get('@old_sound_rabbit_mq.notification_producer')' ?

Reply:
You can inject the NotificationProducerService instead of using $this->get()

Reply:
That's better